New Integration: Pliant (#605)
DataDog/integrations-extras: pliant/README.md

- Clear disk / reprovision storage
- Provision aditional VMs or container nodes in response to load.
- De-comission resources when load is low

## Setup

### Installation

Create one or more workflows that you would like to trigger from a Datadog notification

### Configuration

#### Pliant Setup
**Create a Pliant API Key.**
1. Log in to Pliant and click on your username at the top right of the screen to reveal a menu. Click "API Keys".

![API Key Menu step1][8]

2. From the API keys screen, click "+ Create" at the top right of the screen and title your new API key. Click save and note the the API key will be added to the table.

![Create API Key step2][9]

**Create a Pliant workflow to trigger from Datadog**

1. Navigate to the workflows tab in Pliant. Click "+ Create"  and "Create Flow" to create a new workflow. Title the workflow in the popup and click "Create" to launch the editor into our new workflow.

![Create Flow step1-a-][10]

2. Populate the flow with actions to take upon receiving the Datadog trigger. 

In this example, we named the workflow "RestartHost" and will trigger a host restart from the information Datadog will pass to this workflow when it is triggered.

>   - Add Input variables which will be populated with data sent from Datadog.  Click the "Expand" icon on at the start of the workflow to open the Variable pannel.  We need to create matching **Input** variables. Set all of these input variables to equal empty quotes: `""`. By default, Datadog sends information for the following 

Reviewer: ```suggestion
  - To add Input variables which populated with data sent from Datadog, click the "Expand" icon on at the start of the workflow to open the Variable panel.  To create matching **Input** variables, set all of these input variables to equal empty quotes: `""`. By default, Datadog sends information for the following data:
```

`body`
`last_updated`
`event_type`
`title`
`date`
`org`
`id`

![Expand][11]

> 3. In the expand section, we can also see the the URL of the webhook for the workflow.  To get the endpoint of our Pliant workflow, which we will use to trigger from Datadog with an HTTP request, click the "Expand" icon at the start of the workflow.

Reviewer: ```suggestion
3. To get the endpoint of the Pliant workflow, used to trigger from Datadog with an HTTP request, click the "Expand" icon at the start of the workflow.
```

> Click "cUrl" > "Temporary Bearer Token" and instead select the API key we just created.

Reviewer: ```suggestion
Click "cURL" > "Temporary Bearer Token" and select the API key you just created.
```

![curl][12]

![select key][13]

> We will see our endpoint just after the CURL command, as the endpoint before the CURL command starting with "https". It should resemble: ***https://<YOUR_PLIANT_INSTANCE>/api/v1/trigger/<YOUR_PLIANT_USERNAME>/User/<PATH_TO_WORKFLOW>/<WORKFLOW_NOW>?sync=true&api_key=<YOUR_API_KEY>***

Reviewer: ```suggestion
Your endpoint looks like this: ***https://<YOUR_PLIANT_INSTANCE>/api/v1/trigger/<YOUR_PLIANT_USERNAME>/User/<PATH_TO_WORKFLOW>/<WORKFLOW_NOW>?sync=true&api_key=<YOUR_API_KEY>***
```

![endpoint][14]

Copy this endpoin, starting with ***https*** and ending with the full API Key. Do not include any quotes.

#### Datadog setup
1. Open Datadog and from the left sidebar, click to **Integrations** > **Integrations**.
![integrations][15]

2. Enter "webhooks" in the search bar and click on the **webhooks** entry to reveal a configuration window.
![webhookSearch][16]

> 3. Scroll to "webhooks". Click **New** to add a new webhook to link to our Pliant workflow. First, give the webhook a name in the "name" field. We will name ours *RestartHost* to corespond with our workflow.

Reviewer: ```suggestion
3. Scroll to "webhooks". Click **New** to add a new webhook to link to the Pliant workflow. First, give the webhook a name in the "name" field. This example uses the name *RestartHost*.
```

![webhooksConfig2][17]

> Now, we will paste the URL we copied from our workflow from step 4 of "Create a Pliant workflow to trigger from Datadog". It will resemble: 

Reviewer: ```suggestion
Now, paste the URL copied from step 4. For example: 
```

***https://<YOUR_PLIANT_INSTANCE>/api/v1/trigger/<YOUR_PLIANT_USERNAME>/User/<PATH_TO_WORKFLOW>/<WORKFLOW_NOW>?sync=true&api_key=<YOUR_API_KEY>***

Paste this into the ***URL*** field of the webhook form.

![webhookForm][18]

> The request payload will be pre-configured. Check the "ENCODE AS FORM" box and click save.

Reviewer: ```suggestion
The request payload is pre-configured. Check the "ENCODE AS FORM" box and click save.
```

> Add this integration to any alert notification in Datadog by adding the recipient of `@webhook-RestartHost`. When the monitor triggers an alert, the webhook will trigger your Pliant workflow, and the input variables will be sent to Pliant from Datadog.

Reviewer: ```suggestion
Add this integration to any alert notification in Datadog by adding the recipient of `@webhook-RestartHost`. When the monitor triggers an alert, the webhook triggers your Pliant workflow, and the input variables are sent to Pliant from Datadog.
```

# Agent Check: pliant

## Overview

Pliant.io enhances Datadog notifications with low-code automated workflows, creating a true close-loop-automation solution. This can help with troubleshooting, diagnosis and automated remediation.  

> For more integration information, check out the [Pliant][1] site..

Reviewer 2: ```suggestion
For more integration information, check out the [Pliant][1] site.
```

Examples include:

- Service restart
- Load balancer configuration
- System provisioning
- Clear disk / reprovision storage

> - Provision aditional VMs or container nodes in response to load.

Reviewer 2: ```suggestion
- Provision additional VMs or container nodes in response to load
```
The other list items don't end in `.`

> - De-comission resources when load is low

Reviewer 2: ```suggestion
- De-commission resources when load is low
```

Author: Hello, I wanted to verify that we have completed all requested changes– you mentioned that we should add the integration to "integrations-extras/.github/CODEOWNERS" with the maintainers handle, but I noticed we do not have write access to that file. Is this a change to be made on our end?

Thank you,

Best,
Xavier

Reviewer 3: You should include that change in this PR

Author: Hello, I have made the requested changes to the CODEOWNERS file. If there are any other changes you would like to see, do let me know and I will integrate them.

Thank you!

Xavier

## Troubleshooting

Need help? Contact [Datadog support][7].

[1]: https://pliant.io/
[2]: https://docs.datadoghq.com/agent/kubernetes/integrations/
[3]: https://github.com/DataDog/integrations-core/blob/master/pliant/datadog_checks/pliant/data/conf.yaml.example
[4]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
[5]: https://docs.datadoghq.com/agent/guide/agent-commands/#agent-status-and-information
[7]: https://docs.datadoghq.com/help/
[8]: https://www.pliant.io/wp-content/uploads/2020/05/step1.png
[9]: https://www.pliant.io/wp-content/uploads/2020/05/step2.png
[10]: https://www.pliant.io/wp-content/uploads/2020/05/step1-a-.png
[11]: https://www.pliant.io/wp-content/uploads/2020/05/expand.png
[12]: https://www.pliant.io/wp-content/uploads/2020/05/curl.png
[13]: https://www.pliant.io/wp-content/uploads/2020/05/selectDDkey.png
[14]: https://www.pliant.io/wp-content/uploads/2020/05/endpoint.png
[15]: https://www.pliant.io/wp-content/uploads/2020/05/integrations_.png
[16]: https://www.pliant.io/wp-content/uploads/2020/05/webhook_Search.png
[17]: https://www.pliant.io/wp-content/uploads/2020/05/webhooksConfig3.png

> [18]: https://www.pliant.io/wp-content/uploads/2020/05/webhookForm.png

Reviewer 2: Could we move these images into an `/pliant/images/` directory within this repo and replace the links with raw github versions?
See example: https://github.com/DataDog/integrations-extras/tree/master/k6/images

The externally linked images may not be rendered correctly in our Integrations documentation pages which are generated from the README (https://docs.datadoghq.com/integrations/k6/). This would also be easier to maintain in the future :)

Author: I have added the pliant/images directory in our forked repo and updated the raw image file paths to correspond to the example given. Please verify the addition of the images folder in this pull request along with the updated paths. Images should populate in README upon merge, as file paths will exist upon merge.
